x/stakeibc/keeper/hooks.go

```diff
@@ -184,46 +184,42 @@ func (k Keeper) UpdateRedemptionRates(ctx sdk.Context, depositRecords []recordst
 	}
 }
 
-// Determine the undelegated balance from the deposit records queued for staking
-func (k Keeper) GetUndelegatedBalance(hostZone types.HostZone, depositRecords []recordstypes.DepositRecord) sdk.Dec {
-	// filter to only the deposit records for the host zone with status DELEGATION_QUEUE
-	undelegatedDepositRecords := utils.FilterDepositRecords(depositRecords, func(record recordstypes.DepositRecord) (condition bool) {
-		return ((record.Status == recordstypes.DepositRecord_DELEGATION_QUEUE ||
-			record.Status == recordstypes.DepositRecord_DELEGATION_IN_PROGRESS) &&
-			record.HostZoneId == hostZone.ChainId)
-	})
-
-	// sum the amounts of the deposit records
+// Determine the deposit account balance, representing native tokens that have been deposited
+// from liquid stakes, but have not yet been transferred to the host
+func (k Keeper) GetDepositAccountBalance(chainId string, depositRecords []recordstypes.DepositRecord) sdk.Dec {
+	// sum on deposit records with status TRANSFER_QUEUE or TRANSFER_IN_PROGRESS
 	totalAmount := sdkmath.ZeroInt()
-	for _, depositRecord := range undelegatedDepositRecords {
-		totalAmount = totalAmount.Add(depositRecord.Amount)
+	for _, depositRecord := range depositRecords {
+		transferStatus := (depositRecord.Status == recordstypes.DepositRecord_TRANSFER_QUEUE ||
```

Comment:
What about the opposite direction? When someone wants to redeem and we have to undelegate (21 days) an then send the tokens from the host zone back over to the stride chain... aren't there going to be similar transfer queue states for that case until the tokens are given to the original user? Or do we take the stToken immediately so everything after the unbonding is triggered it isn't an ammount we need to care about in these totals?

Reply:
we never transfer the LSM tokens back to Stride. When they are detokenized, the NFT is burned and the record is deleted

We do take the stTokens immediately during unbonding